lib/score_board.rb

```diff
@@ -0,0 +1,31 @@
+# calculates overall scoring
+class ScoreBoard
+  class << self
+    def total_score(game_frames = frames)
+      self.frames = game_frames
+      frame_scores.sum
+    end
+
+    def scores_and_rolls
+      frames.map { |frame| { rolls: frame.rolls, score: frame.score } }
+    end
+
+    def running_total
+      scores = []
+      i = 0
+      while i < frames.count
+        scores << frame_scores[0..i].sum
+        i += 1
+      end
+      scores
+    end
```

Comment:
This feels inefficient as you are always counting back again from the start. Maybe you could store the accumulated score and add only the last frame each time?

Reply:
I see what you mean. spent about 10 minutes last night on this and I did manage to refactor the method to be shorter:
```
def running_total
  frame_scores.map.with_index { |frame, i| frame_scores[0..i].sum }
end
```

although it still accumulates in the same way, so I think it's still inefficient. I'll try again if I find the time,